### PR TITLE
Add:Guide about using HTTP Caching in Tailcall.

### PR DESCRIPTION
**Pull request by xonx4l**
issue-:#90

/claim #90

Guide about using HTTP Caching in Tailcall.

### CODE REVIEW

File: docs/guides/HTTP-Caching.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+### Caching Headers
+Tailcall respects caching headers like Cache-Control sent by the upstream API. This allows the upstream API to control what is cached and for how long.
+
+For example, if the upstream response contains:
+
+```Cache-Control: max-age=300
+```
+Tailcall will only cache that response for 300 seconds.
+
+This allows the upstream API to control caching behavior.
```
Comment (tusharmath) [inline docs/guides/HTTP-Caching.md:53]: Tailcall also supports expiry header

File: docs/guides/HTTP-Caching.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+### Cache Configuration
+Tailcall uses a Least Recently Used (LRU) cache by default. You can configure the cache size and max age in the Tailcall config file:
+
+To enable HTTP caching globally:
+
+```upstream: 
+  baseURL: https://api.example.com
+  httpCache: true
+```
+You can also configure the cache size and max age:
```
Comment (tusharmath) [inline docs/guides/HTTP-Caching.md:36]: We don't have this feature. Can you raise an issue for it. We will add it

File: docs/guides/HTTP-Caching.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+->Repeated API calls for unchanged data
+->APIs with caching headers indicating cacheable data
+->It avoids unnecessary calls to upstream APIs in these cases.
+
+### Summary
+->Enable with httpCache: true under upstream
+->Configure cache size and max age
+->Respects upstream caching headers
+->Caches identical requests during cache lifetime
+->Improves performance by reducing API calls
```
Comment (tusharmath) [inline docs/guides/HTTP-Caching.md:69]: Expand on the article by talking about cacheControlHeader in server

File: docs/guides/HTTP-Caching.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+### Cache Configuration
+Tailcall uses a Least Recently Used (LRU) cache by default. You can configure the cache size and max age in the Tailcall config file:
+
+To enable HTTP caching globally:
+
+```upstream: 
+  baseURL: https://api.example.com
+  httpCache: true
+```
```
Comment (tusharmath) [inline docs/guides/HTTP-Caching.md:35]: Update all examples with with Tailcall DSL

**Review by tusharmath — CHANGES_REQUESTED** [review]
There is a bit of duplicate content that can be reduced

File: docs/guides/HTTP-Caching.md (hunk 4 above)
Comment (xonx4l) [reply docs/guides/HTTP-Caching.md:35]: @schema
@server(
  port: 4000
)
@upstream(
  baseURL: "https://api.example.com"
  httpCache: true
) {
  query: Query
}.    is this looks good to you @tusharmath .

Comment (tusharmath) [reply docs/guides/HTTP-Caching.md:35]: Yes.

**Pull request merged**